woocommerce/woocommerce-ios
# Dynamic Dashboard: Most recent orders - Load order statuses from remote

Part of: #12792 


## Description


Loads the order statuses from remote for the "Most recent orders" dashboard card. 

Changes
- Loads orders statuses from remote for the "Most recent orders" dashboard card. 
- Saves and restores status selection.
- Adds unit tests for the card's view model.

Pending work related to this "Most recent orders" dashboard card is tracked at #12792

## Testing instructions


- Create a JN site with custom order statuses. I used this plugin to create orders with custom statuses. https://wordpress.org/plugins/custom-order-statuses-woocommerce/
- Enable the feature flag `dynamicDashboardM2` and build and login into the app.
- Enable the "Most recent orders" card on the dashboard.
- Confirm that the card now displays the 3 most recent orders. If there are no orders available empty view should be displayed.
- Change the status filter and validate the orders are listed based on the selected status. Observe that status picker also displays the custom order status. 

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/524475/ea73634e-64da-454f-961d-a186f71363c1" width="320"/>


---
